src/module-loader.ts

```diff
@@ -136,11 +136,15 @@ export default class ModuleLoader {
     let isESM: boolean
     let filePath: string
 
+    const isPlugin = (config: IConfig|IPlugin): config is IPlugin => {
+      return (<IPlugin>config).type !== undefined
+    }
+
     try {
       filePath = require.resolve(modulePath)
       isESM = ModuleLoader.isPathModule(filePath)
     } catch {
-      filePath = Config.tsPath(config.root, modulePath)
+      filePath = isPlugin(config) ? Config.tsPath(config.root, modulePath, config.type) : Config.tsPath(config.root, modulePath)
```

Comment:
pass `plugin.type` to `Config.tsPath` if `config` is a plugin.

hooks get loaded here:
https://github.com/oclif/core/blob/main/src/config/config.ts#L310

usually these are paths to compiled JS files like this one:
https://github.com/salesforcecli/plugin-info/blob/main/package.json#L93

Linked plugins are supposed to get compiled at runtime by ts-node but hooks were being skipped because the `plugin.type` wasn't being passed to `tsPath`:
https://github.com/oclif/core/blob/main/src/config/ts-node.ts#L98

commands from linked plugins are loaded here (see `plugin.type` is being passed):
https://github.com/oclif/core/blob/main/src/config/plugin.ts#L182